hub/hub_service.bal

```diff
@@ -21,18 +21,81 @@ import kafkaHub.security;
 import kafkaHub.persistence as persist;
 import kafkaHub.config;
 import kafkaHub.util;
+import kafkaHub.healthcheck;
+import ballerina/jballerina.java;
+import kafkaHub.connections as conn;
+import ballerinax/kafka;
+
+http:Service healthCheckService = service object {
+    
+    isolated resource function get .() returns healthcheck:HealthCheckResp {
+        //diskspace
+        handle handleStr = java:fromString(config:CURRENT_WORKING_DIR);
+        handle fileObj = newFile(java:fromString(getCurrent(handleStr).toString()));
+        int usableSpace = getUsableSpace(fileObj);
+        int totalSpace = getTotalSpace(fileObj);
+        healthcheck:DiskSpaceMetaData diskSpaceMetaData = {free: usableSpace,total: totalSpace};
+        healthcheck:HealthCheckResp diskSpace={status: "UP",details: {diskSpaceMetaData}};
+        
+        //kafka
+        string producerStatus ="DOWN"; 
+        string consumerStatus="DOWN"; 
+        string kafkaStatus="DOWN";
+        kafka:TopicPartition[]|kafka:Error producerResult =  conn:statePersistProducer->getTopicPartitions(config:HEALTH_CHECK_TOPIC);
+        if(producerResult is kafka:TopicPartition[]){
+            producerStatus = "UP";
+        }
+        kafka:TopicPartition[]|kafka:Error result =  conn:healthCheckConsumer->getTopicPartitions(config:HEALTH_CHECK_TOPIC);
+        if(result is kafka:TopicPartition[]){
+            consumerStatus = "UP";
+        }
```

Comment:
In both producer and consumer I see only getTopicPartitions being called, what is the point in having separate producer and consumer ?

Reply:
using just producer to check connectivity.

Reply:
Resolved.